src/adaptation/precond.jl

```diff
@@ -2,50 +2,59 @@
 #### Robust online (co-)variance estimators.
 ####
 
+## Variance estimator
+
 abstract type VarEstimator{T} end
 
 # NOTE: this naive variance estimator is used only in testing
-mutable struct NaiveVar{T} <: VarEstimator{T}
+mutable struct NaiveVar{T<:AbstractVector{<:AbstractVecOrMat{<:AbstractFloat}}} <: VarEstimator{T}
     n :: Int
-    S :: Vector{Vector{T}}
+    S :: T
 end
 
 NaiveVar(::Type{T}=Float64) where {T} = NaiveVar(0, Vector{Vector{T}}())
-NaiveVar(::Type{T}, ::Int) where {T} = NaiveVar(T)
-NaiveVar(::Int) = NaiveVar(Float64)
+NaiveVar(::Type{T}, sz::Tuple{Int}) where {T} = NaiveVar(0, Vector{Vector{T}}())
+NaiveVar(::Type{T}, sz::Tuple{Int,Int}) where {T} = NaiveVar(0, Vector{Matrix{T}}())
+NaiveVar(sz::Tuple{Vararg{Int}}) = NaiveVar(Float64, sz)
```

Comment:
Can you clarify via a comment what happens when `sz` has more than 2 elements?

Reply:
It should throw an error because this passing a tuple of size 3 is not supported.

Reply:
I see. Maybe consider simplifying the code into

```
NaiveVar(sz::Tuple{Int}, ::Type{T}) where {T} = NaiveVar(0, Vector{Vector{T}}())
NaiveVar(sz::Tuple{Int,Int}, ::Type{T}) where {T} = NaiveVar(0, Vector{Matrix{T}}())
```

That is, swapping the order of the arguments and remove the `NaiveVar(sz::Tuple{Vararg{Int}}) ` definition since it does not add any new stuff.

Reply:
I decide to remove all the construcotrs which are only for convenient purpose.